src/Microsoft.NetCore.Analyzers/Core/Performance/DoNotUseCountWhenAnyCanBeUsed.cs

```diff
@@ -274,13 +300,34 @@ public void AnalyzeBinaryOperation(OperationAnalysisContext context)
             {
                 var binaryOperation = (IBinaryOperation)context.Operation;
 
-                if (binaryOperation.IsComparisonOperator() &&
-                    (IsLeftCountComparison(binaryOperation, out var methodName) || IsRightCountComparison(binaryOperation, out methodName)))
+                if (binaryOperation.IsComparisonOperator())
                 {
+                    string operation;
+
+                    if (IsLeftCountComparison(binaryOperation, out var methodName, out var shouldNegate))
+                    {
+                        operation = OperationBinaryLeft;
+                    }
+                    else if (IsRightCountComparison(binaryOperation, out methodName, out shouldNegate))
+                    {
+                        operation = OperationBinaryRight;
+                    }
+                    else
+                    {
+                        return;
+                    }
+
+                    var propertiesBuilder = ImmutableDictionary.CreateBuilder<string, string>(StringComparer.Ordinal);
```

Comment:
We have a PooledDictionary that you might be able to use for the builder.

Reply:
What would be the benefit of that?

Reply:
It is pooled, but it is fine not to use it given this is not a hot code path.